nav2_costmap_2d/src/collision_checker.cpp

```diff
@@ -0,0 +1,209 @@
+// Copyright (c) 2019 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "nav2_costmap_2d/collision_checker.hpp"
+#include "nav2_costmap_2d/cost_values.hpp"
+#include "nav2_util/line_iterator.hpp"
+#include "nav2_costmap_2d/exceptions.hpp"
+#include "nav2_costmap_2d/footprint.hpp"
+
+namespace nav2_costmap_2d
+{
+
+CollisionChecker::CollisionChecker(
+  rclcpp::Node::SharedPtr ros_node,
+  std::shared_ptr<CostmapSubscriber> costmap_sub,
+  std::shared_ptr<FootprintSubscriber> footprint_sub,
+  tf2_ros::Buffer & tf_buffer,
+  std::string name)
+: tf_buffer_(tf_buffer),
+  node_(ros_node), name_(name),
+  costmap_sub_(costmap_sub), footprint_sub_(footprint_sub)
+{
+  node_->get_parameter_or<std::string>(
+    "global_frame", global_frame_, std::string("map"));
+  node_->get_parameter_or<std::string>(
+    "robot_base_frame", robot_base_frame_, std::string("base_link"));
+}
+
+CollisionChecker::~CollisionChecker() {}
+
+bool CollisionChecker::isCollisionFree(
+  const geometry_msgs::msg::Pose2D & pose)
+{
+  try {
+    if (scorePose(pose) < 0) {
+      return false;
+    }
+    return true;
+  } catch (const IllegalPoseException & e) {
+    RCLCPP_ERROR(node_->get_logger(), "%s", e.what());
+    return false;
+  } catch (const CollisionCheckerException & e) {
+    RCLCPP_ERROR(node_->get_logger(), "%s", e.what());
+    return false;
+  }
+}
+
+double CollisionChecker::scorePose(
+  const geometry_msgs::msg::Pose2D & pose)
+{
+  Costmap2D * costmap_;
```

Comment:
Why have a costmap pointer? Maybe move the `worldToMap` checks to `CostmapSubscriber`

Reply:
Because the intention of the `CostmapSubscriber` was to convert the msg into the `Costmap2D` object to access its API calls. I think it's better not to duplicate that code in the subscriber class